### PR TITLE
#161966618 add notifications

### PR DESCRIPTION
**Pull request by TeamoreA**
### What does this pull request do?
 This PR implements getting notifications for users.
### Description of the tasks to be completed?
- User get notifications when a user they follow creates an article
- A user gets notified when a comment is added on an article they have favourited.
- Get all notifications endpoint
  `GET /api/notifications/`
- Mark notification as read.
  `PUT /api/notifications/<pk>`
- delete notification
  `DELETE /api/notifications/<pk>`
- Users can receive email notifications
- Users can opt in or out of getting notifications in their profiles

### How should this be manually tested?
       $ git clone https://github.com/andela/ah-technocrats.git
       $ cd ah-technocrats
       $ git checkout ft-notifications-161966618
       $ virtualenv -p python3 env
       $ source venv/bin/activate
       $ pip install -r requirements.txt
       $ python manage.py makemigrations
       $ python manage.py migrate
       $ python manage.py runserver

testing: 
       $ ./manage.py test

### Screenshots:
Get all notifications
image

Mark notification as read
image

Delete a notification
image

### What are the relevant Pivotal Tracker Stories
[#161966618](https://www.pivotaltracker.com/story/show/161966618)
# Checklist:
- [x] My code follows the style guidelines of this project
- [ ] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] My PR has one commit.

### CODE REVIEW

**Review by verenceLola — DISMISSED** [review]
@TeamoreA and @actlikewill there is a grammar issue with your return message after marking a notification as read. Please check that.

File: authors/apps/articles/extra_views.py [hunk 1]
```diff
@@ -69,4 +78,77 @@ def list(self, request):
                     'message': 'No articles bookmarked yet.'
                 },
                 status=status.HTTP_200_OK
-            )
\ No newline at end of file
+            )
+
+
+class ArticleAPIView(APIView):
```
Comment (Mnickii) [inline authors/apps/articles/extra_views.py]: Why did you move this from the main articles View?

**Review by Mnickii — CHANGES_REQUESTED** [review]

**Review by salma-nyagaka — COMMENTED** [review]
Please edit your message in the second screenshot

File: authors/apps/articles/extra_views.py (hunk 1 above)
Comment (actlikewill) [reply authors/apps/articles/extra_views.py]: The original views file was over the recommended 250 lines long. So to avoid making it too long while adding our changes we moved it to the new extra views file. Codeclimate recomends doing this.

Comment (Mnickii) [reply authors/apps/articles/extra_views.py]: If this is the problem you aim to solve, the better way would be to create a views folder and have every model's view in its own file in the folder.

Comment (actlikewill) [reply authors/apps/articles/extra_views.py]: Ok, Should I go ahead and do that within this PR?

Comment (Mnickii) [reply authors/apps/articles/extra_views.py]: Yeah.

**Comment by Mnickii** [comment]
Rebase

**Review by Mnickii — APPROVED** [review]
LGTM
